src/lib/util.ts

```diff
@@ -0,0 +1,249 @@
+  useApi,
+} from '@operate-first/probot-kubernetes';
+import { operationsTriggered } from './counters';
+import { IncomingMessage } from 'http';
+import { Context } from 'probot';
+import * as k8s from '@kubernetes/client-node';
+import { HttpError } from '@kubernetes/client-node';
+import { InstallationAccessTokenAuthentication } from '@octokit/auth-app';
+
+// Simple callback wrapper - executes an async operation and based on the result it inc() operationsTriggered counted
+export const wrapOperationWithMetrics = async (
+  callback: Promise<any>,
+  labels: { install: number | undefined; method: string }
+) => {
+  const response = await callback
+    .then(() => ({
+      status: 'Succeeded',
+    }))
+    .catch(() => ({
+      status: 'Failed',
+    }));
+
+  const optick = (status: string) => {
+    operationsTriggered
+      .labels({
+        ...labels,
+        status,
+        operation: 'k8s',
+      })
+      .inc();
+  };
+
+  optick(response.status);
+};
+
+export declare type IssueFormPipelineParams = {
+  SOURCE_REPO: string;
+  TARGET_REPO: string;
+  ISSUE_NUMBER: string;
+  PAYLOAD: string;
+  TASK_TYPE: string;
+  SCRIPT_PATH: string;
+  WORKING_DIR: string;
+  WORKING_BRANCH_PREFIX: string;
+  APP_USER_ID: string;
+  APP_SLUG: string;
+};
+
+export const createPipelineRun = async (
+  name: string,
+  issuePipeline: IssueFormPipelineParams,
+  context: Context<'issue_comment.created'> | Context<'issues.opened'>
+): Promise<{ response: IncomingMessage; body: object }> => {
+  const params = [
+    {
+      name: 'SOURCE_REPO',
+      value: issuePipeline.SOURCE_REPO,
+    },
+    {
+      name: 'TARGET_REPO',
+      value: issuePipeline.TARGET_REPO,
+    },
+    {
+      name: 'ISSUE_NUMBER',
+      value: issuePipeline.ISSUE_NUMBER,
+    },
+    {
+      name: 'PAYLOAD',
+      value: issuePipeline.PAYLOAD,
+    },
+    {
+      name: 'TASK_TYPE',
+      value: issuePipeline.TASK_TYPE,
+    },
+    {
+      name: 'SCRIPT_PATH',
+      value: issuePipeline.SCRIPT_PATH,
+    },
+    {
+      name: 'SECRET_NAME',
+      value: getTokenSecretName(context),
+    },
+    {
+      name: 'WORKING_DIR',
+      value: issuePipeline.WORKING_DIR,
+    },
+    {
+      name: 'WORKING_BRANCH_PREFIX',
+      value: issuePipeline.WORKING_BRANCH_PREFIX,
+    },
+    {
+      name: 'APP_USER_ID',
+      value: issuePipeline.APP_USER_ID,
+    },
+    {
+      name: 'APP_SLUG',
+      value: issuePipeline.APP_SLUG,
+    },
+  ];
+
+  const pipelineRunpayload = {
+    apiVersion: 'tekton.dev/v1beta1',
+    kind: 'PipelineRun',
+    metadata: {
+      generateName: name + '-',
+    },
+    spec: {
+      pipelineRef: {
+        name: 'issue-form-pipeline',
+      },
+      params,
+      workspaces: [
+        {
+          name: 'utility-scripts',
+          configMap: {
+            name: 'utility-scripts',
+            defaultMode: 110,
+          },
+        },
+        {
+          name: 'shared-data',
+          emptyDir: {},
```

Comment:
This doesn't work. Empty dir is always empty per each pod. Since you're using it as a workspace in a pipeline, this doesn't carry data over to the next pod in the pipeline. You need to use PVCs in Pipelines.

https://tekton.dev/docs/pipelines/workspaces/#overview
![image](https://user-images.githubusercontent.com/7453394/200553525-5b6116fb-09e2-440a-85e4-ba817699b54e.png)

Reply:
@tumido sure, but we're not passing data between tasks, only between steps. The tasks still need the taskruns (created by the pipeline runs) to make the emptydir workspace available for the individual tasks. AFAIK I still need to specify the emptydir workspace within the pipelinerun for this.

Reply:
Ok, I didn't know that. :slightly_smiling_face: 

In that case, do you really need a workspace? Why don't you just define a volume in the `Task`? But it works either way, this is just a bit confusing to me... If you don't need parametrized workspace, why to use it?

Reply:
I think that's a fair point, it shouldn't be confusing -- I'll add the volume in the task instead.

Reply:
done @tumido can you re-review